File: app/src/main/java/com/morh/management/wrappers/PagedResponse.java
package com.morh.management.wrappers;

import android.net.Uri;

import java.util.List;

public class PagedResponse<T> extends Response<T>
{
    public int PagedNumber;

    public int PageSize;

    public Uri FirstPage;

    public Uri LastPage;

    public int TotalPages;

    public int TotalRecords;

    public Uri NextPage;

    public Uri PreviousPage;

    public PagedResponse() { }

    public PagedResponse(List<T> data, int pageNumber, int pageSize)
    {
        PagedNumber = pageNumber;
        PageSize = pageSize;
        Data = data;
        Message = null;
        Succeeded = true;
        Errors = null;
    }
}

File: app/src/main/java/com/morh/management/wrappers/PagedResponse.kt
package com.morh.management.wrappers

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import java.lang.reflect.Array

@Serializable
data class PagedResponse<T> (

    @SerialName("PagedNumber")
    val PagedNumber: Int,

    @SerialName("PageSize")
    val PageSize: Int,

    @SerialName("FirstPage")
    val FirstPage: String,

    @SerialName("LastPage")
    val LastPage: String,

    @SerialName("TotalPages")
    val TotalPages: Int,

    @SerialName("TotalRecords")
    val TotalRecords: Int,

    @SerialName("NextPage")
    val NextPage: String,

    @SerialName("PreviousPage")
    val PreviousPage: String? = null,

    @SerialName("Data")
    val Data: List<T>?,

    @SerialName("Succeeded")
    val Succeeded: Boolean,

    @SerialName("Errors")
    val Errors: String? = null,

    @SerialName("Message")
    val Message: String? = null,

    @SerialName("Code")
    val Code: String? = null
)
